LatinumNetwork/mosaic
# added breadcrumbs component

## Purpose/Description:

This PR introduces enhancements to the Breadcrumbs component, expanding its capabilities to support a variety of child components such as text, links, buttons, and icons. These changes aim to improve the flexibility and user experience of the component.

### Screenshots:
![Captura de pantalla 2023-10-10 a la(s) 11 46 47 a m](https://github.com/LatinumNetwork/mosaic/assets/45240616/7bbca3d8-9b26-45d5-964b-76db73835ff5)


## Jira Link:

Link to the relevant Jira issue or task. (e.g., [BRAN-154](https://collagegroup.atlassian.net/browse/BRAN-154?atlOrigin=eyJpIjoiZWFmMzNiMWJmZTc1NGNhMWE5NjZkOTIzYTA3NzBjNzMiLCJwIjoiaiJ9))

## Test Plan:

Test Scenario 1: Ensure basic breadcrumbs display correctly.

Steps:
1. Navigate to the component's Storybook page.
2. Access the "Basic" story.
3. Observe the displayed breadcrumbs.
Expected Result: 
Breadcrumbs should be displayed in a linear fashion, with clear legible text.

Test Scenario 2: Verify icon support in breadcrumbs.
Steps:
1. Navigate to the component's Storybook page.
2. Access the "With Icons" story.
3. Observe the displayed breadcrumbs.
Expected Result: Breadcrumbs items should include icons where specified, aligned properly with the text.

## Additional Notes

### Reviewer Instructions:

- Focus on the responsiveness of the breadcrumbs, especially when icons are included.
- Check the compatibility of the new changes with existing usages of the Breadcrumbs component in the application.
- Validate the appearance in both light and dark modes.



[BRAN-154]: https://collagegroup.atlassian.net/browse/BRAN-154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
